wp-content/themes/twentytwenty-child/functions.php

```diff
@@ -293,4 +338,220 @@ function update_students_settings() {
+}
+add_action( 'manage_student_posts_custom_column', 'print_extra_columns' );
+
+/**
+ * Make the custom column sortable
+ */
+function sort_student_status( $columns ) {
+    $columns['Active'] = __('status');
+    return $columns;
+}
+add_filter( 'manage_edit-student_sortable_columns', 'sort_student_status' );
+
+/**
+ * Define how to order our sortable custom column
+ */
+function student_status_orderby( $query ) {
+    if( ! is_admin() ) return;
+ 
+    $orderby = $query->get( 'orderby' );
+ 
+    if( 'status' == $orderby ) {
+        $query->set('meta_key','status');
+        $query->set('orderby','meta_value');
+    }
+}
+add_action( 'pre_get_posts', 'student_status_orderby' );
+
+/**
+ * Function executed by AJAX to update activity status
+ */
+function update_student_status(){
+    $status = 0;
+    if( esc_attr( $_POST['checked'] ) === 'true' ) $status = 1;
+    update_post_meta( sanitize_text_field( $_POST['student-id'] ), 'status', $status );
+}
+add_action( 'wp_ajax_nopriv_update_student_status', 'update_student_status' );
+add_action( 'wp_ajax_update_student_status', 'update_student_status' );
+
+/** 
+ * Add custom top-level menu page 
+ * */
+function register_dictionary_menu() {
+    add_menu_page(
+        'Dictionary',
+        'Dictionary',
+        'manage_options',
+        'dictionary',
+        'dictionary_callback' );
+}
+add_action( 'admin_menu', 'register_dictionary_menu' );
+
+/**
+ * Callback func for the menu page
+ * */
+function dictionary_callback() {
+    echo '<div> <h2>Dictionary</h2> </div>';
+    settings_fields( 'dictionary' );
+    do_settings_sections( 'dictionary' );
+}
+
+/** 
+ * Add custom settings to the new settings page 
+ * */
+function register_dictionary_settings() {
+    add_settings_section( 'dictionary', '', null, 'dictionary' );
+
+    add_settings_field( 'dictionary_search'   , 'Search up a word: ', 'show_dictionary_search'  , 'dictionary', 'dictionary' );
+
+    register_setting( 'dictionary', 'dictionary_search' );
+}
+add_action( 'admin_init', 'register_dictionary_settings' );
+
+/** 
+ * Callback for a custom setting 
+ * */
+function show_dictionary_search() {
+    $search = isset($_POST['dictionary-search']) ? sanitize_text_field($_POST['dictionary-search']) : '';
+    echo 
+    '<form method="post" action="' . admin_url('admin.php?page=dictionary') . '"> 
+        <input type="search" class="dictionary-search" name="dictionary-search" placeholder="Search..." value="' . $search . '"> 
+        <input type="submit" class="dictionary-submit"> 
+    </form> <div class="result-data"> </div>'; 
+}
+
+/** 
+ * Function executed by AJAX to remotely get data from Oxford Dictionary
+ * */
+function search_oxford_dictionary(){
+
+    $result = wp_remote_get( 'https://www.oxfordlearnersdictionaries.com/definition/english/' . sanitize_title_with_dashes( $_POST['word'] ) );
+    echo $result['body'];
```

Comment:
We echo the result right away? What if the site is down? when we do a 3rd party requests, we need to make sure:
* the data is there
* the data is escaped and sanitized
* the requeest won't kill our site

Reply:
Hmm, he did add additional stuff but I will let you decide if we should close this one.